### PR TITLE
test: add test for verify attestation

### PR DESCRIPTION
**Pull request by SweeXordious**
## Overview

Closes https://github.com/celestiaorg/quantum-gravity-bridge/issues/134

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: src/test/QuantumGravityBridge.t.sol [hunk 1]
```diff
@@ -93,6 +94,50 @@ contract RelayerTest is DSTest {
         assertEq(bridge.state_dataRootTupleRoots(nonce), newTupleRoot);
     }
 
+    function testVerifyAttestation() public {
+        uint256 initialVelsetNonce = 0;
+        // data root tuple root nonce.
+        uint256 nonce = 1;
+        // commitment to a set of roots.
+        bytes32 newTupleRoot = 0x82dc1607d84557d3579ce602a45f5872e821c36dbda7ec926dfa17ebc8d5c013;
```
Comment (adlerjohn) [inline src/test/QuantumGravityBridge.t.sol:102]: can you document how these roots and sidenodes were constructed?

File: src/test/QuantumGravityBridge.t.sol [hunk 2]
```diff
@@ -93,6 +94,50 @@ contract RelayerTest is DSTest {
         assertEq(bridge.state_dataRootTupleRoots(nonce), newTupleRoot);
     }
 
+    function testVerifyAttestation() public {
+        uint256 initialVelsetNonce = 0;
+        // data root tuple root nonce.
+        uint256 nonce = 1;
+        // commitment to a set of roots.
```
Comment (SweeXordious) [inline src/test/QuantumGravityBridge.t.sol:101]:
```suggestion
        // commitment to a set of roots.
        // these values were generated using the tendermint implementation of binary merkle trees:
        // https://github.com/celestiaorg/celestia-core/blob/60310e7aa554bb76b735a010847a6613bcfe06e8/crypto/merkle/proof.go#L33-L48
```

Comment (adlerjohn) [reply src/test/QuantumGravityBridge.t.sol:101]: Yes, but also what are the leaf values?

File: src/test/QuantumGravityBridge.t.sol [hunk 3]
```diff
@@ -93,6 +94,50 @@ contract RelayerTest is DSTest {
         assertEq(bridge.state_dataRootTupleRoots(nonce), newTupleRoot);
     }
 
+    function testVerifyAttestation() public {
```
Comment (SweeXordious) [inline src/test/QuantumGravityBridge.t.sol:97]:
```suggestion
    /*
    the values used in the verify attestation test are in the format `<height padded to 32 bytes || data root>`, which
    represent an encoded `abi.encode(DataRootTuple)`:

    0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
    0x00000000000000000000000000000000000000000000000000000000000000010101010101010101010101010101010101010101010101010101010101010101
    0x00000000000000000000000000000000000000000000000000000000000000020202020202020202020202020202020202020202020202020202020202020202
    0x00000000000000000000000000000000000000000000000000000000000000030303030303030303030303030303030303030303030303030303030303030303
    */
    function testVerifyAttestation() public {
```

Comment (SweeXordious) [reply src/test/QuantumGravityBridge.t.sol:97]: @adlerjohn something like this?

Comment (adlerjohn) [reply src/test/QuantumGravityBridge.t.sol:97]: Yeah, looks good

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**